src/main/kotlin/fi/metatavu/metaform/server/controllers/AdminThemeController.kt

```diff
@@ -0,0 +1,76 @@
+package fi.metatavu.metaform.server.controllers
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import fi.metatavu.metaform.server.persistence.dao.AdminThemeDAO
+import fi.metatavu.metaform.server.persistence.model.AdminTheme
+import java.util.UUID
+import javax.enterprise.context.ApplicationScoped
+import javax.inject.Inject
+
+/**
+ * Controller for admin themes
+ */
+@ApplicationScoped
+class AdminThemeController {
+    @Inject
+    lateinit var adminThemeDAO: AdminThemeDAO
+
+    @Inject
+    lateinit var mapper: ObjectMapper
+
+    /**
+     * Creates a new admin theme
+     *
+     * @param id the id of the theme
+     * @param data the data in the theme
+     * @param name the name of the theme
+     * @param slug the slug of the theme
+     * @param creatorId who created this theme
+     * @param lastModifierId who last modified this theme
+     *
+     * @return new [AdminTheme] object
+     */
+    fun create(
+        id: UUID,
+        data: Any?,
+        name: String,
+        slug: String,
```

Comment:
Change all the slug as nullable in controller, keep the slug as notnull in JPA.
In create admin theme, validate the slug with regex "^[a-z\\d]+(?:[-, _][a-z\\d]+)*$" and check for duplicates. if user provided slug is valid (not duplicated and matches regex), use the slug. if not valid, if not return e.g. invalidSlugException, catch it in api layer and return 400. if the slug is null, create the slug prefix from the name using slugify. if the slug prefix is not duplicated, use the prefix, else loop through all the numbers to check if e.g. slug-1 is duplicated, if not, use the slug